### ISSUE TITLE
Block rule not working for request going through AdGuard

### ISSUE THREAD

**Issue by Kusresa** [AdguardTeam/AdguardForAndroid#1778]
I am filtering the AdGuard app and have Swiftkey keyboard installed and for some reason, a Swiftkey telemetry server request goes through the app. Even worse, I tried to create block rules to stop this request yet when it shows up in the filtering log it isn't red and never gets blocked. The request is not very common but still happens multiple times throughout the week. Please look into this!

**Comment by ameshkov** [comment]
@Kusresa could you please take a screenshot of the issue? I don't fully understand what's going on

**Comment by Kusresa** [comment]
@ameshkov 

I have the Swiftkey Keyboard app installed and turned off all internet data access to it using AdGuard. But I see a request in the filtering log that shows as going through the AdGuard app (which I'm filtering) which is to a swiftkey server (url is `telemetry.api.swiftkey.com`). This is weird because no request should show up for swiftkey in the filtering log since I've blocked internet access to its app but for some reason it is there because the request is somehow shown as going through the AdGuard app. Second, I've tried creating blocks for the request like `||swiftkey.com^$important` but they don't work and the swiftkey server request keeps showing up in the log as unblocked.

**Comment by ameshkov** [comment]
It looks as if the app detection fails to detect it properly.

Could you please check it on the latest nightly build?
https://agrd.io/android_nightly

**Comment by Kusresa** [comment]
Forgot about this but to let you know I uninstalled swiftkey can't check anymore. I do think I saw another app doing this though and the request going through Adguard was as at the start of the filtering log after I opened the Adguard app. Not sure if this is still happening but I'll keep a check on it. 

Also I noticed in Android 8 on latest stable Adguard version a lot of requests that were indicated as coming from the proper app is now just coming from "Android OS". Is this a bug or because of Android 8?
